Ct 3915 setup new sar ir case type

willmcb opened the pull request:
## Description
It adds:
- `Case::SAR::InternalReview` inherited from the standard Sar
- Workflow for internal review
- Tests for new model
- Policy object + placeholder test file for this if there is more to add to it.
- Data migration for new correspondence type
- Some factories (Note commented out code here will likely be useful in further iterations so I have kept it in for now)


## Self-review checklist
<!-- Action these things before requesting reviews -->
* [ ] (1) Quick stakeholder demo done OR
* [ ] (2) ...bug with before and after screenshots
* [ ] (3) Tests passing
* [ ] (4) Branch ready to be merged (not work in progress)
* [ ] (5) No superfluous changes in diff
* [ ] (6) No TODO's without new ticket numbers
* [ ] (7) PR Prefixed with ticket number e.g. `CT-7654 ...`
* [ ] (8) Data migration script is created if any of letter templates is changed

### Related JIRA tickets
[CT-3927](https://dsdmoj.atlassian.net/browse/CT-3927)

Commit:
Add feature flag for internal review

Commit:
Add SAR_INTERNAL_REVIEW correspondence type

Commit:
Add internal review policy classes + spec

Commit:
Add Case::SAR::InternalReview case type

Essentially this is a copy of the standard SAR as the case is so
similar. Have commented out the model tests that are failing due
to the lack of a workflow for this new case type.

Commit:
Add new workflows for SAR internal review and uncomment previously breaking specs

Commit:
Add data migration for internal review type

Commit:
add internal review to base policy list

Commit:
Fix spec

ymao2 reviewed (commented):
You may need to add the new case type into 00_preamble.yml file too. 
Nice job 👍 , couple of  comments,  happy to discuss them

Commit:
Add convenience method

Adds convenience method that returns true if the case type is
Case::SAR:InternalReview

Commit:
Reuse SAR workflow for SAR Internal Review cases

As the workflows are so similar, it makes sense to reuse them rather
than creating new workflows.

Adds class method state_machine_name to Case::SAR::InternalReview which
adds points to the existing SAR workflow, and deletes the Internal
Review workflows that are no longer needed.

Commit:
Adapt predicate to SAR IR case

Commit:
move #is_sar_internal_review? to base class

Commit:
Add tests for sar correspondence type

Commit:
new wrapper method to reduce cyclomatic complexity
